fix: suggestions API issue

bietkul opened the pull request:
<!--
Work-in-progress PRs are welcome as a way to get early feedback - just prefix
the title with [WIP].

Add the change in the changelog (except for test changes and docs updates).
Please edit CHANGELOG.md and add the change under the appropriate category (NEW
FEATURES, IMPROVEMENTS & BUG FIXES) along with the PR number.
-->

#### What does this do / why do we need it?

- Show Distinct Issue

https://www.loom.com/share/2aa51be18d684c38ba8c55807620af5b

- User ID preference

https://www.loom.com/share/785998e40694498089d3f60417c1b839

- Predictive Suggestions Issues

https://www.loom.com/share/98630bdad21f447181ab4e5c26302f4a

Fix
https://www.loom.com/share/5750f0f51a9e42b08a6dc5362652ead7

#### What should your reviewer look out for in this PR?

#### Which issue(s) does this PR fix?

#### If this PR affects any API reference documentation, please share the updated endpoint references

<!--

- [ ] I've updated the doc.

Doc link shared here - <Updated API Reference Doc link>

-->

<!--

fixes #
fixes #

-->

bietkul committed:
feat: add support for storedquery cat

bietkul committed:
Merge branch 'dev' of github.com:appbaseio/reactivesearch-api into dev

bietkul committed:
Merge branch 'dev' of github.com:appbaseio/reactivesearch-api into dev

bietkul committed:
Merge branch 'dev' of github.com:appbaseio/reactivesearch-api into dev

bietkul committed:
Merge branch 'dev' of github.com:appbaseio/reactivesearch-api into dev

bietkul committed:
Merge branch 'dev' of github.com:appbaseio/reactivesearch-api into dev

bietkul committed:
Merge branch 'dev' of github.com:appbaseio/reactivesearch-api into dev

bietkul committed:
Merge branch 'dev' of github.com:appbaseio/reactivesearch-api into dev

bietkul committed:
Merge branch 'dev' of github.com:appbaseio/reactivesearch-api into dev

bietkul committed:
feat: add support for IP restrictions for users

bietkul committed:
Merge branch 'dev' of github.com:appbaseio/reactivesearch-api into dev

bietkul committed:
Merge branch 'dev' of github.com:appbaseio/reactivesearch-api into dev

bietkul committed:
Merge branch 'dev' of github.com:appbaseio/reactivesearch-api into dev

bietkul committed:
fix: show distinct suggestions

bietkul committed:
fix: use userId with preference

bietkul committed:
fix: test cases

bietkul committed:
fix: predictive suggestions

bietkul committed:
Merge branch 'fix/3-dec' of github.com:appbaseio/reactivesearch-api into fix/3-dec

siddharthlatest commented on plugins/querytranslate/suggestions.go:
```diff
@@ -816,14 +814,29 @@ func getFinalSuggestions(config SuggestionsConfig, rawHits []ESDoc) []Suggestion
+func SanitizeString(str string) string {
+	// remove extra spaces
+	s := str
+	specialChars := []string{"[", "-", "+", ".", "^", ":", ",", "]"}
```
```suggestion
	specialChars := []string{"'", "/", "[", "-", "+", ".", "^", ":", ",", "]"}
```

siddharthlatest committed:
fix(query translate):
- use userIP for preference fallback if userID isn't set
- add other special characters that should be ignored from comparison
